ssl/ssl_sess.c

```diff
@@ -1161,7 +1161,7 @@ void SSL_CTX_flush_sessions(SSL_CTX *s, long t)
      */
     while (s->session_cache_tail != NULL) {
         current = s->session_cache_tail;
-        if (t == 0 || sess_timedout((time_t)t, current)) {
+        if (t == 0 || sess_timedout(ossl_time_from_time_t(t), current)) {
```

Comment:
It might be more efficient to do this conversion (`ossl_time_from_time_t()`) before the loop.